client/my-sites/email/email-providers-stacked-comparison/provider-cards/provider-card-props.ts

```diff
@@ -0,0 +1,63 @@
+import { Domain } from '@automattic/data-stores/dist/types/site';
+import { TranslateResult } from 'i18n-calypso';
+import { ReactElement } from 'react';
+
+export type Site = {
+	ID: number;
+	slug: string;
+};
+
+export interface ProviderCard {
+	additionalPriceInformation?: TranslateResult;
+	badge?: ReactElement;
+	buttonLabel?: TranslateResult;
+	children?: ReactElement;
+	description: TranslateResult;
+	detailsExpanded: boolean;
+	discount?: ReactElement | null;
+	expandButtonLabel: TranslateResult;
+	features: TranslateResult[];
+	footerBadge?: ReactElement;
+	formattedPrice?: TranslateResult;
+	formFields?: ReactElement;
+	isDomainEligibleForTitanFreeTrial?: boolean;
+	logo: ReactElement;
+	onExpandedChange: ( providerKey: string, expanded: boolean ) => void;
+	onButtonClick?: ( event: React.MouseEvent ) => void;
+	productName: TranslateResult;
+	providerKey: string;
+	showExpandButton: boolean;
+}
+
+export type EmailProvidersStackedCardProps = {
+	comparisonContext: string;
+	cart?: any;
+	cartDomainName?: string;
+	selectedSite?: Site;
+	currencyCode?: string;
+	currentRoute?: string;
+	domain?: any;
+	domainName?: string;
+	domainsWithForwards?: Domain[];
+	gSuiteProduct?: string;
+	hasCartDomain?: boolean;
+	isGSuiteSupported?: boolean;
+	productsList?: string[];
+	requestingSiteDomains?: boolean;
+	selectedDomainName: string;
+	shoppingCartManager?: any;
+	source: string;
+	titanMailProduct?: any;
+};
+
+type ValueError = {
+	value: string;
+	error: string;
+};
+
+export type Mailbox = {
+	uuid: string;
+	domain: ValueError;
+	mailbox: ValueError;
+	password: ValueError;
+};
```

Comment:
Don't we have these types available from the validation code? I don't know if we have it for Titan or G Suite, but I seem to recall there being TS in at least one of those locations.

Reply:
GSuite has a type for it, Titan have a prop type definition for JS